src/test/java/net/sf/jabref/importer/ImportFormatReaderTest.java

```diff
@@ -0,0 +1,67 @@
+package net.sf.jabref.importer;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+
+@RunWith(Parameterized.class)
+public class ImportFormatReaderTest {
+
+    private ImportFormatReader reader;
+
+    private final String resourceName;
+    private final int count;
+    public final String format;
+
+
+    public ImportFormatReaderTest(String resource, String format, int count) {
+        this.resourceName = resource;
+        this.format = format;
+        this.count = count;
+    }
+
+    @Before
+    public void setUp() {
+        Globals.prefs = JabRefPreferences.getInstance();
+        reader = new ImportFormatReader();
+        reader.resetImportFormats();
```

Comment:
Yes. There is no explicit constructor. Maybe has something to do with the
preferences not being initialized? resetImportFormat() is called quite
early in JabRef.start